OneCommunityGlobal/HGNRest
# Shereen purchase equipment

# Description
Adds functionality for the user to place purchase order for equipment. [/bmdashboard/equipment/purchase/] (http://localhost:3000/bmdashboard/equipment/purchase/).

Implements # (WBS) 9.2.2

![Screenshot 2024-05-09 105828](https://github.com/OneCommunityGlobal/HGNRest/assets/7041610/8fd6bbf7-0f5b-4f8a-8e25-8180d026c5d0)



## Related PRS (if any):
This backend PR is related to the #2251 front-end PR.
To test this backend PR you need to checkout https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2251 frontend PR.
…

## Main changes explained:
- Added controller function for equipment Purchase.
- Added Routing for purchase equipment request
…

## How to test:
1. check into current branch.
2. Run npm install and npm run build && npm start to run the backend locally.
3. Clear site data/cache
4. Log in as any user
5. Go to /bmdashboard (http://localhost:3000/bmdashboard) and log in again.
6. Navigate to [/bmdashboard/equipment/purchase] (http://localhost:3000/bmdashboard/equipment/purchase/)
7. Place a purchase order for an equipment.
8. Verify that the green success message has been displayed after the form submission.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/7041610/7f3fd264-1599-4385-85c5-995b541c0344
